jwql/website/apps/jwql/monitor_pages/monitor_readnoise_bokeh.py

```diff
@@ -59,102 +72,128 @@ def load_data(self):
+                                      self.readnoise_histogram),
+                         title=self.aperture)
 
-        # Update the mean readnoise figures for all amps
+    def plot_readnoise_amplifers(self):
+
+        self.amp_plots = []
         for amp in ['1', '2', '3', '4']:
-            readnoise_vals = np.array([getattr(result, 'amp{}_mean'.format(amp)) for result in self.query_results])
-            self.refs['mean_readnoise_source_amp{}'.format(amp)].data = {'time': expstarts,
-                                                                         'time_iso': expstarts_iso,
-                                                                         'mean_rn': readnoise_vals,
-                                                                         'filename': filenames,
-                                                                         'nints': nints,
-                                                                         'ngroups': ngroups}
-            self.refs['mean_readnoise_figure_amp{}'.format(amp)].title.text = 'Amp {}'.format(amp)
-            self.refs['mean_readnoise_figure_amp{}'.format(amp)].hover.tooltips = [('file', '@filename'),
-                                                                                   ('time', '@time_iso'),
-                                                                                   ('nints', '@nints'),
-                                                                                   ('ngroups', '@ngroups'),
-                                                                                   ('readnoise', '@mean_rn')]
-
-            # Update plot limits if data exists
-            if len(readnoise_vals) != 0:
-                self.refs['mean_readnoise_xr_amp{}'.format(amp)].start = expstarts.min() - timedelta(days=3)
-                self.refs['mean_readnoise_xr_amp{}'.format(amp)].end = expstarts.max() + timedelta(days=3)
-                min_val, max_val = min(x for x in readnoise_vals if x is not None), max(x for x in readnoise_vals if x is not None)
-                if min_val == max_val:
-                    self.refs['mean_readnoise_yr_amp{}'.format(amp)].start = min_val - 1
-                    self.refs['mean_readnoise_yr_amp{}'.format(amp)].end = max_val + 1
-                else:
-                    offset = (max_val - min_val) * .1
-                    self.refs['mean_readnoise_yr_amp{}'.format(amp)].start = min_val - offset
-                    self.refs['mean_readnoise_yr_amp{}'.format(amp)].end = max_val + offset
-
-    def update_readnoise_diff_plots(self):
-        """Updates the readnoise difference image and histogram"""
+
+            amp_plot = figure(title='Amp {}'.format(amp), width=280, height=280, x_axis_type='datetime')
+            amp_plot.xaxis[0].ticker.desired_num_ticks = 4
+
+            readnoise_vals = np.array([getattr(result, 'amp{}_mean'.format(amp)) for result in self.db.query_results])
```

Comment:
What if self.db.query_results is empty?

Reply:
Hmm I bokeh seems to be smart enough to know how to handle empty list like objects or `None`. I can make a little step here to just create an empty numpy array just in case.